packages/common/src/GRPCClient.ts

```diff
@@ -305,6 +305,78 @@ export default class ConcordiumNodeClient {
             .response;
         return Buffer.from(response.value).toString('hex');
     }
+
+    /**
+     * Gets a stream of finalized blocks.
+     *
+     * @param abortSignal an AbortSignal to close the stream. Note that the
+     * stream does not close itself as it is infinite, so usually you'd want
+     * to provide this parameter.
+     * @returns An AsyncIterator stream of finalized blocks.
+     */
+    getFinalizedBlocks(
+        abortSignal?: AbortSignal
+    ): AsyncIterable<v1.FinalizedBlockInfo> {
+        const opts = { abort: abortSignal };
+        const blocks = this.client.getFinalizedBlocks(v2.Empty, opts).responses;
+        return mapAsyncIterable(blocks, translate.commonBlockInfo);
+    }
+
+    /**
+     * Gets a stream of blocks. To get a stream of only finalized blocks
+     * use `getFinalizedBlocks()` instead.
+     *
+     * @param abortSignal an AbortSignal to close the stream. Note that the
+     * stream does not close itself as it is infinite, so usually you'd want
+     * to provide this parameter.
+     * @returns An AsyncIterator stream of blocks.
+     */
+    getBlocks(abortSignal?: AbortSignal): AsyncIterable<v1.ArrivedBlockInfo> {
+        const opts = { abort: abortSignal };
+        const blocks = this.client.getBlocks(v2.Empty, opts).responses;
+        return mapAsyncIterable(blocks, translate.commonBlockInfo);
+    }
+
+    /**
+     * Waits until given transaction is finalized.
+     *
+     * @param transactionHash a transaction hash as a bytearray.
+     * @param timeoutTime the number of milliseconds until the function throws error.
+     * @returns A blockhash as a byte array.
+     */
+    async waitForTransactionFinalization(
+        transactionHash: HexString,
+        timeoutTime?: number
+    ): Promise<HexString> {
+        if (timeoutTime) {
+            setTimeout(() => {
+                throw Error('Function timed out.');
+            }, timeoutTime);
```

Comment:
This doesn't actually make the function throw. (It throws in a different context)
Also you should use clearTimeout if the the item finalized before the timeout.